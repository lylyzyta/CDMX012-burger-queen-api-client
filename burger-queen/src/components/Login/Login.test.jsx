/* eslint-disable no-undef */
import React from 'react'
// import { BrowserRouter as Router } from 'react-router-dom'
// import { create, act } from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import LoginPage from './Login'

const mockedUsedNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate
}))

describe(('Test component Login'), () => {
  test('render component Login', () => {
    const { asFragment } = render(<LoginPage />)
    const loginButton = screen.getByText('Login')
    expect(asFragment()).toMatchSnapshot()
    expect(loginButton).toBeDefined()
  })

  test('value when click on button login', () => {
    const mockOnClick = jest.fn()
    render(<LoginPage onClick={mockOnClick}/>)
    const button = screen.getByText('Login')
    fireEvent.click(button)
    expect(mockOnClick).toHaveBeenCalledTimes(0)
  })
})
