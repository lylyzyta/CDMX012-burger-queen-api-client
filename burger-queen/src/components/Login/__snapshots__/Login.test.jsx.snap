// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Login render component Login 1`] = `
<DocumentFragment>
  <div
    class="container-login"
  >
    <form
      class="register-container"
    >
      <img
        alt="logo-icon"
        class="logo-icon"
        src="Logo_Image.png"
      />
      <h1
        class="title-h1"
      >
         Welcome!
      </h1>
      <h3
        class="title-h3"
      >
        Login to your account
      </h3>
      <div
        class="box-form"
      >
        <img
          alt="email-icon"
          class="icon"
          src="emailIcon.svg"
        />
        <input
          class="input-form"
          name="email"
          placeholder="email@example.com"
          type="email"
        />
      </div>
      <div
        class="box-form"
      >
        <img
          alt="password-icon"
          class="icon"
          src="passwordIcon.svg"
        />
        <input
          class="input-form"
          id="password"
          name="password"
          placeholder="Password..."
          type="password"
        />
      </div>
      <button
        class="btn-form"
        type="button"
      >
         Login
      </button>
      <div
        class="title-error"
      />
    </form>
  </div>
</DocumentFragment>
`;
